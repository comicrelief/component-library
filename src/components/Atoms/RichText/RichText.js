import React from 'react';
import styled from 'styled-components';
import PropTypes from 'prop-types';

export const Inner = styled.div`
  text-align: ${props => props.align};
`;

const RichText = ({ align = 'left', markup = '', ...rest }) => (
  <Inner
    align={align}
    dangerouslySetInnerHTML={{ __html: markup }}
    {...rest}
  />
);

RichText.propTypes = {
  align: PropTypes.oneOf(['left', 'center', 'right']),
  markup: PropTypes.string
};

export default RichText;
