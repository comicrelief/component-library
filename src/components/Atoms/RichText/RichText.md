```js
<RichText
  alignment={'center'}
  markup={'<p>Some <strong>strong</strong> test markup</p>'}
></RichText>
```
