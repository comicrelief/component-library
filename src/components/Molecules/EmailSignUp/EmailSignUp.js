import React, { useState } from 'react';
import PropTypes from 'prop-types';
import Text from '../../Atoms/Text/Text';
import Link from '../../Atoms/Link/Link';
import {
  ESUWrapper,
  TopCopyWrapper,
  Form,
  PrivacyCopyWrapper,
  InputField,
  ButtonWrapper
} from './EmailSignUp.style';

const EmailSignUp = ({
  title,
  topCopy,
  isSuccess,
  successCopy,
  errorMsg,
  privacyCopy,
  subscribe,
  buttonColor,
  backgroundColor,
  ...rest
}) => {
  const [value, setValue] = useState('');
  const subscriptionForm = (
    <Form>
      <InputField
        name="email"
        id="email"
        hasAria={false}
        showLabel={false}
        type="email"
        errorMsg={errorMsg}
        label="email"
        placeholder="example@youremail.com"
        value={value}
        onChange={event => setValue(event.target.value)}
      />
      <ButtonWrapper backgroundColor={backgroundColor}>
        <Link
          type="button"
          color={buttonColor}
          as="button"
          href="/#"
          onClick={() => subscribe(value)}
        >
          Subscribe
        </Link>
      </ButtonWrapper>
    </Form>
  );

  const privacyContainer = (
    <PrivacyCopyWrapper>{privacyCopy}</PrivacyCopyWrapper>
  );
  return (
    <ESUWrapper backgroundColor={backgroundColor} {...rest}>
      {title && <Text tag="h1">{title}</Text>}
      <TopCopyWrapper>{isSuccess ? successCopy : topCopy}</TopCopyWrapper>
      {!isSuccess && subscriptionForm}
      {!isSuccess && privacyContainer}
    </ESUWrapper>
  );
};

EmailSignUp.propTypes = {
  /** title */
  title: PropTypes.string,
  /** top copy */
  topCopy: PropTypes.node.isRequired,
  /** displayed copy when subscription is successful */
  successCopy: PropTypes.node.isRequired,
  /** boolean if true display successCopy */
  isSuccess: PropTypes.bool.isRequired,
  /** email error message */
  errorMsg: PropTypes.string.isRequired,
  /** privacy copy */
  privacyCopy: PropTypes.node.isRequired,
  /** subscription function */
  subscribe: PropTypes.func.isRequired,
  /** background color */
  backgroundColor: PropTypes.string,
  /** button color */
  buttonColor: PropTypes.string
};

EmailSignUp.defaultProps = {
  title: null,
  backgroundColor: 'deep_violet',
  buttonColor: 'red'
};

export default EmailSignUp;
