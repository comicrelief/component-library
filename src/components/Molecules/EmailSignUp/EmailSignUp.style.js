import styled from 'styled-components';

import spacing from '../../../theme/shared/spacing';
import Input from '../../Atoms/Input/Input';
import Text from '../../Atoms/Text/Text';

const ESUWrapper = styled.div`
  display: flex;
  flex-direction: column;
  font-size: ${({ theme }) => theme.fontSize('s')};
  padding: ${spacing('l')};
  color: ${({ theme }) => theme.color('white')};
  background-color: ${({ theme, backgroundColor }) =>
    theme.color(backgroundColor)};
`;

const TopCopyWrapper = styled.div`
  display: flex;
  margin-bottom: ${spacing('none')};
`;

const ButtonWrapper = styled.div`
  margin-top: ${spacing('md')};
  button {
    border-color: ${({ theme, backgroundColor }) =>
      theme.color(backgroundColor)};
    width: 100%;
    font-size: ${({ theme }) => theme.fontSize('s')};
    @media ${({ theme }) => theme.breakpoint('small')} {
      width: auto;
    }
  }
`;

const PrivacyCopyWrapper = styled.div`
  display: flex;
  flex-direction: column;
  a {
    cursor: pointer;
    padding: 0 ${spacing('xsm')};
    color: ${({ theme }) => theme.color('white')};
    border-color: ${({ theme }) => theme.color('white')};
    :hover {
      color: ${({ theme }) => theme.color('white')};
      border-color: ${({ theme }) => theme.color('white')};
    }
  }
`;

const Form = styled.form`
  display: flex;
  flex-direction: column;
  margin-bottom: ${spacing('md')};
`;

const InputField = styled(Input)`
  width: 100%;
`;

const Title = styled(Text)`
  margin-bottom: ${spacing('l')};
  font-size: ${({ theme }) => theme.fontSize('xxl')};

  @media ${({ theme }) => theme.breakpoint('medium')} {
    font-size: ${({ theme }) => theme.fontSize('big')};
  }
`;

export {
  ESUWrapper,
  TopCopyWrapper,
  PrivacyCopyWrapper,
  ButtonWrapper,
  Form,
  InputField,
  Title
};
