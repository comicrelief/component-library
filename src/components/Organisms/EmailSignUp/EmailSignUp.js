import React, { useState } from 'react';
import PropTypes from 'prop-types';

import Select from '../../Atoms/SelectField/SelectField';
import Text from '../../Atoms/Text/Text';
import {
  ESUWrapper,
  TopCopyWrapper,
  Form,
  PrivacyCopyWrapper,
  InputField,
  ButtonWrapper,
  Title
} from './EmailSignUp.style';
import { Button } from '../Membership/Membership.style';

const EmailSignUp = ({
  title,
  topCopy,
  isSuccess,
  isSchools,
  schoolsCopy,
  selectItems,
  successCopy,
  errorMsg,
  privacyCopy,
  subscribe,
  validate,
  buttonColor,
  backgroundColor,
  ...rest
}) => {
  const [value, setValue] = useState('');
  const [age, setAge] = useState('');
  const [next, setNext] = useState(false);

  const handleSubmit = e => {
    // Just stopping weird things happening for now when the user submits with the enter key
    // rather than the Submit button
    e.preventDefault();
  };

  const schoolsSelect = (
    <>
      <Select
        hideLabel
        label="Please choose an option"
        placeholder="Please choose an option"
        description="Please choose an option"
        errorMsg={errorMsg}
        options={selectItems}
        value={age}
        onChange={event => setAge(event.target.value)}
      />
      <Text tag="p" weight="400" size="m">
        {schoolsCopy}
      </Text>
    </>
  );

  const subscriptionSchoolsForm = (
    <Form onSubmit={e => handleSubmit(e)}>
      {!next && (
        <>
          <InputField
            aria-label="Email address"
            name="email"
            id="email"
            hasAria={false}
            showLabel={false}
            type="email"
            errorMsg={errorMsg}
            label="email"
            placeholder="example@youremail.com"
            value={value}
            onChange={event => setValue(event.target.value)}
          />
          <ButtonWrapper backgroundColor={backgroundColor}>
            <Button
              as="input"
              type="submit"
              data-test="subscribe-button-school"
              color={buttonColor}
              onClick={() => validate({ email: value }) && setNext(true)}
              value=" Subscribe"
            />
          </ButtonWrapper>
        </>
      )}
      {next && (
        <>
          {schoolsSelect}
          <ButtonWrapper backgroundColor={backgroundColor}>
            <Button
              as="input"
              type="submit"
              data-test="subscribe-button-school"
              color={buttonColor}
              onClick={() =>
                validate({ email: value, age }) &&
                subscribe({ email: `${value}`, age: `${age}` })
              }
              value=" Subscribe"
            />
          </ButtonWrapper>
        </>
      )}
    </Form>
  );

  const subscriptionForm = (
    <Form onSubmit={e => handleSubmit(e)}>
      <InputField
        aria-label="Email address"
        name="email"
        id="email"
        hasAria={false}
        showLabel={false}
        type="email"
        errorMsg={errorMsg}
        label="email"
        placeholder="example@youremail.com"
        value={value}
        onChange={event => setValue(event.target.value)}
      />
      <ButtonWrapper backgroundColor={backgroundColor}>
        <Button
          as="input"
          data-test="subscribe-button"
          type="submit"
          color={buttonColor}
          onClick={() => validate({ email: value }) && subscribe(value)}
          value=" Subscribe"
        />
      </ButtonWrapper>
    </Form>
  );

  const privacyContainer = (
    <PrivacyCopyWrapper>{privacyCopy}</PrivacyCopyWrapper>
  );

  return (
    <ESUWrapper backgroundColor={backgroundColor} {...rest}>
      <Title tag="h2" size="xxl" weight="400" family="Anton" uppercase>
        {title}
      </Title>
      {(!next || isSuccess) && (
        <TopCopyWrapper>{isSuccess ? successCopy : topCopy}</TopCopyWrapper>
      )}
      {!isSuccess && (isSchools ? subscriptionSchoolsForm : subscriptionForm)}
      {privacyContainer}
    </ESUWrapper>
  );
};

EmailSignUp.propTypes = {
  /** title */
  title: PropTypes.string.isRequired,
  /** top copy */
  topCopy: PropTypes.node.isRequired,
  /** displayed copy when subscription is successful */
  successCopy: PropTypes.node.isRequired,
  /** boolean if true display successCopy */
  isSuccess: PropTypes.bool.isRequired,
  /** boolean if true display schools dropdown */
  isSchools: PropTypes.bool,
  /** email error message */
  errorMsg: PropTypes.string.isRequired,
  /** privacy copy */
  privacyCopy: PropTypes.node.isRequired,
  /** privacy copy */
  schoolsCopy: PropTypes.node,
  selectItems: PropTypes.PropTypes.arrayOf(
    PropTypes.shape({
      value: PropTypes.string.isRequired,
      displayValue: PropTypes.string.isRequired
    })
  ),
  /** subscription function */
  subscribe: PropTypes.func.isRequired,
  /** validation function */
  validate: PropTypes.func.isRequired,
  /** background color */
  backgroundColor: PropTypes.string,
  /** button color */
  buttonColor: PropTypes.string
};

EmailSignUp.defaultProps = {
  backgroundColor: 'deep_violet_dark',
  buttonColor: 'red',
  isSchools: false,
  schoolsCopy: '',
  selectItems: []
};

export default EmailSignUp;
