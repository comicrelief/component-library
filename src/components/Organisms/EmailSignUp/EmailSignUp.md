# Email SignUp Form

```js
// EmailSignUpForm.js is just a wrapper for viewing in the CL
// The email sign up component that is used/exported is _EmailSignUp.js
import EmailSignUpForm from './EmailSignUpForm';
import Text from '../../Atoms/Text/Text';

<>
    <Text tag="p">This EmailSignUpForm component exists purely to show the EmailSignUp component functioning within the Component Library; applications are to provide their own react-hook-form form and validation, based on these.</Text>

    <EmailSignUpForm />
</>
```
