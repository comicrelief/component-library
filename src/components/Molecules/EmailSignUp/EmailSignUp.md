# Email Sign Up

```js
import RichText from '../../Atoms/RichText/RichText';

const title = 'Stay in the know!';
const topCopy = (
  <RichText
    markup={`<p>Get regular email updates and info on what we're up to!</p>`}
  />
);
const privacyCopy = (
  <RichText
    markup={`<p>Our <a class="link link--white inline" href="/privacy-notice">Privacy Policy</a> describes how we handle and protect your information.<br><br>If you are under 18, please make sure you have your parents’ permission before providing us with any personal details.</p>`}
  />
);
const successCopy = (
  <RichText
    markup={`<p>Thanks! Your first email will be with you shortly</p>`}
  />
);

initialState = { isSuccess: false, errorMsg: '' };

const sendEmail = email => {
  setTimeout(() => setState({ isSuccess: !state.isSuccess }), 2000);
  console.log(email);
};

const validate = ({ email }) => {
  let isValid = false;
  if (email.includes('@')) {
    isValid = true;
    setState({ errorMsg: '' });
  } else {
    setState({ errorMsg: 'invalid email!' });
  }
  return isValid;
};

<EmailSignUp
  title={title}
  topCopy={topCopy}
  successCopy={successCopy}
  isSuccess={state.isSuccess}
  privacyCopy={privacyCopy}
  errorMsg={state.errorMsg}
  subscribe={sendEmail}
  validate={validate}
/>;
```

# Email Sign Up Schools

```js
import RichText from '../../Atoms/RichText/RichText';

const title = 'Stay in the know!';
const topCopy = (
  <RichText
    markup={`<p>Get regular email updates and info on what we're up to!</p>`}
  />
);
const privacyCopy = (
  <RichText
    markup={`<p>Our <a class="link link--white inline" href="/privacy-notice">Privacy Policy</a> describes how we handle and protect your information.<br><br>If you are under 18, please make sure you have your parents’ permission before providing us with any personal details.</p>`}
  />
);
const successCopy = (
  <RichText
    markup={`<p>Thanks! Your first email will be with you shortly</p>`}
  />
);
const selectItems = [
  { value: '', displayValue: '-- Select age group --' },
  { value: 'Option one', displayValue: 'The first option' },
  {
    value: 'Option two',
    displayValue: 'The second option'
  },
  { value: 'Option three', displayValue: 'The third option' },
  { value: 'Option four', displayValue: 'The fourth option' }
];

initialState = { isSuccessSchools: false, errorMsg: '' };

sendEmail = emailAndAge => {
  setTimeout(
    () => setState({ isSuccessSchools: !state.isSuccessSchools }),
    2000
  );
  console.log(emailAndAge);
};

const validate = ({ email, age }) => {
  let isValid = false;
  if (email.includes('@')) {
    isValid = true;
    setState({ errorMsg: '' });
  } else {
    setState({ errorMsg: 'invalid email!' });
  }
  if (isValid === true && typeof age !== 'undefined') {
    if (age) {
      setState({ errorMsg: '' });
    } else {
      isValid = false;
      setState({ errorMsg: 'invalid age!' });
    }
  }
  return isValid;
};

<EmailSignUp
  title={title}
  topCopy={topCopy}
  successCopy={successCopy}
  schoolsCopy="Now please select your teaching group so you get the right updates."
  isSuccess={state.isSuccessSchools}
  selectItems={selectItems}
  isSchools
  privacyCopy={privacyCopy}
  errorMsg={state.errorMsg}
  subscribe={sendEmail}
  validate={validate}
/>;
```
