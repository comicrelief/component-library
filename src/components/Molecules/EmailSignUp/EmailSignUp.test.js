import React from 'react';
import 'jest-styled-components';

import renderWithTheme from '../../../hoc/shallowWithTheme';
import EmailSignUp from './EmailSignUp';
import RichText from '../../Atoms/RichText/RichText';

it('renders correctly', () => {
  const top =
    '<h1> Top Copy</h1> <p>Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua. Ut enim ad minim veniam, quis nostrud exercitation ullamco laboris nisi ut aliquip ex ea commodo consequat.</p>';
  const success =
    '<h1> Success Copy</h1> <p>Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua. Ut enim ad minim veniam, quis nostrud exercitation ullamco laboris nisi ut aliquip ex ea commodo consequat.</p>';
  const privacy =
    'check <a href="https://www.comicrelief.com/privacy-notice">Privacy policy</a>';
  const tree = renderWithTheme(
    <>
      <EmailSignUp
        topCopy={<RichText markup={top} />}
        successCopy={<RichText markup={success} />}
        isSuccess={false}
        errorMsg=""
        buttonColor="teal"
        privacyCopy={<RichText markup={privacy} />}
        subscribe={() => 'Done'}
      />
      <EmailSignUp
        topCopy={<RichText markup={top} />}
        successCopy={<RichText markup={success} />}
        isSuccess
        errorMsg=""
        backgroundColor="red"
        privacyCopy={<RichText markup={privacy} />}
        subscribe={() => 'Done'}
      />
    </>
  ).toJSON();

  expect(tree).toMatchInlineSnapshot(`
    Array [
      .c1 {
      text-transform: inherit;
    }

    .c7 {
      text-transform: inherit;
      font-weight: bold;
    }

    .c11 {
      display: inline-block;
      padding: 11px 17px;
      text-align: center;
      -webkit-text-decoration: none;
      text-decoration: none;
      font-weight: 700;
      border-radius: 30px;
      cursor: pointer;
      background-color: #22CFD8;
      color: #2C0230;
    }

    .c11:hover {
      background-color: #86E4E9;
      color: #2C0230;
    }

    .c9 {
      box-sizing: border-box;
      width: 100%;
      height: 50px;
      padding: 13px;
      margin: 10px 0;
      font-size: 1.2rem;
      background-color: #FFFFFF;
      border: 1px solid;
      border-color: #2C0230;
      box-shadow: none;
      -webkit-appearance: none;
      -moz-appearance: none;
      appearance: none;
      color: #2C0230;
    }

    .c9:focus {
      border: 1px solid #666;
    }

    .c9:focus::-webkit-input-placeholder {
      color: transparent;
    }

    .c9:focus:-moz-placeholder {
      color: transparent;
    }

    .c9:focus::-moz-placeholder {
      color: transparent;
    }

    .c9:focus:-ms-input-placeholder {
      color: transparent;
    }

    .c5 {
      display: -webkit-box;
      display: -webkit-flex;
      display: -ms-flexbox;
      display: flex;
      -webkit-flex-direction: column;
      -ms-flex-direction: column;
      flex-direction: column;
    }

    .c6 {
      visibility: border:0;
      -webkit-clip: rect(0 0 0 0);
      clip: rect(0 0 0 0);
      -webkit-clip-path: inset(50%);
      clip-path: inset(50%);
      height: 1px;
      margin: -1px;
      overflow: hidden;
      padding: 0;
      position: absolute;
      white-space: nowrap;
      width: 1px;
    }

    .c3 {
      text-align: left;
    }

    .c0 {
      display: -webkit-box;
      display: -webkit-flex;
      display: -ms-flexbox;
      display: flex;
      -webkit-flex-direction: column;
      -ms-flex-direction: column;
      flex-direction: column;
      font-size: 1rem;
      padding: 2rem;
      color: #FFFFFF;
      background-color: #2C0230;
    }

    .c2 {
      display: -webkit-box;
      display: -webkit-flex;
      display: -ms-flexbox;
      display: flex;
      margin-bottom: 0;
    }

    .c10 {
      margin-top: 1rem;
    }

    .c10 button {
      border-color: #2C0230;
      width: 100%;
      font-size: 1rem;
    }

    .c12 {
      display: -webkit-box;
      display: -webkit-flex;
      display: -ms-flexbox;
      display: flex;
      -webkit-flex-direction: column;
      -ms-flex-direction: column;
      flex-direction: column;
    }

    .c12 a {
      cursor: pointer;
      padding: 0 0.25rem;
      color: #FFFFFF;
      border-color: #FFFFFF;
    }

    .c12 a:hover {
      color: #FFFFFF;
      border-color: #FFFFFF;
    }

    .c4 {
      display: -webkit-box;
      display: -webkit-flex;
      display: -ms-flexbox;
      display: flex;
      -webkit-flex-direction: column;
      -ms-flex-direction: column;
      flex-direction: column;
      margin-bottom: 1rem;
    }

    .c8 {
      width: 100%;
    }

    @media (min-width:1440px) {
      .c11 {
        padding: 16px 30px;
      }
    }

    @media (min-width:740px) {
      .c9 {
        max-width: 290px;
      }
    }

    @media (min-width:740px) {
      .c10 button {
        width: auto;
      }
    }

    <div
        className="c0"
      >
        <h1
          className="c1"
          color="inherit"
        />
        <div
          className="c2"
        >
          <div
            className="c3"
            dangerouslySetInnerHTML={
              Object {
                "__html": "<h1> Top Copy</h1> <p>Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua. Ut enim ad minim veniam, quis nostrud exercitation ullamco laboris nisi ut aliquip ex ea commodo consequat.</p>",
              }
            }
          />
        </div>
        <form
          className="c4"
        >
          <label
            className="c5"
            htmlFor="signup-email"
          >
            <span
              className="c6 c7"
              color="inherit"
            >
              
            </span>
            <input
              aria-describedby="signup-email"
              className="c8 c9"
              name="email"
              onChange={[Function]}
              placeholder="example@youremail.com"
              type="email"
              value=""
            />
            
          </label>
          <div
            className="c10"
          >
            <button
              className="c11"
              color="teal"
              href="/#"
              onClick={[Function]}
              target="_self"
              type="button"
            >
              Subscribe
            </button>
          </div>
        </form>
        <div
          className="c12"
        >
          <div
            className="c3"
            dangerouslySetInnerHTML={
              Object {
                "__html": "check <a href=\\"https://www.comicrelief.com/privacy-notice\\">Privacy policy</a>",
              }
            }
          />
        </div>
      </div>,
      .c1 {
      text-transform: inherit;
    }

    .c3 {
      text-align: left;
    }

    .c0 {
      display: -webkit-box;
      display: -webkit-flex;
      display: -ms-flexbox;
      display: flex;
      -webkit-flex-direction: column;
      -ms-flex-direction: column;
      flex-direction: column;
      font-size: 1rem;
      padding: 2rem;
      color: #FFFFFF;
      background-color: #E52630;
    }

    .c2 {
      display: -webkit-box;
      display: -webkit-flex;
      display: -ms-flexbox;
      display: flex;
      margin-bottom: 0;
    }

    <div
        className="c0"
      >
        <h1
          className="c1"
          color="inherit"
        />
        <div
          className="c2"
        >
          <div
            className="c3"
            dangerouslySetInnerHTML={
              Object {
                "__html": "<h1> Success Copy</h1> <p>Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua. Ut enim ad minim veniam, quis nostrud exercitation ullamco laboris nisi ut aliquip ex ea commodo consequat.</p>",
              }
            }
          />
        </div>
      </div>,
    ]
  `);
});
