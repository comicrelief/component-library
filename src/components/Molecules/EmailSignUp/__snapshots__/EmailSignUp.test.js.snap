// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders ESU School correctly 1`] = `
.c1 {
  font-size: 1rem;
  text-transform: uppercase;
  font-weight: 400;
  font-family: 'Anton',Impact,sans-serif;
}

.c5 {
  font-size: 1rem;
  text-transform: inherit;
  font-weight: bold;
  font-family: 'Montserrat',Helvetica,Arial,sans-serif;
}

.c8 {
  font-size: 1.25rem;
  text-transform: inherit;
  font-weight: 400;
  font-family: 'Montserrat',Helvetica,Arial,sans-serif;
}

.c10 {
  display: inline-block;
  padding: 1rem 1.5rem;
  text-align: center;
  -webkit-text-decoration: none;
  text-decoration: none;
  font-weight: 700;
  border-radius: 2rem;
  cursor: pointer;
  background-color: #22CFD8;
  color: #2C0230;
}

.c10:hover {
  background-color: #86E4E9;
  color: #2C0230;
}

.c7 {
  background-color: #FFFFFF;
  width: 100%;
  font-size: 1.25rem;
  display: block;
  width: 100%;
  box-sizing: border-box;
  padding: 6px 10px;
  border-width: 1px;
  border-style: solid;
  border-color: #2C0230;
  margin: 0;
  position: relative;
  -moz-appearance: none;
  -webkit-appearance: none;
  border-radius: 0px;
  padding: 12px 50px 8px 10px;
  height: 50px;
  font-weight: 300;
  background: white url(mock.asset) calc(100% - 20px) 18px/20px 20px no-repeat;
  cursor: pointer;
  margin-bottom: 0;
  cursor: pointer;
  -webkit-appearance: none;
  -moz-appearance: none;
  appearance: none;
  box-shadow: border-box;
  border-radius: 0;
}

.c7:focus {
  border: 1px solid #666;
}

.c4 {
  display: block;
  margin-bottom: 8px;
}

.c6 {
  border: 0;
  -webkit-clip: rect(0 0 0 0);
  clip: rect(0 0 0 0);
  -webkit-clip-path: inset(50%);
  clip-path: inset(50%);
  height: 1px;
  margin: -1px;
  overflow: hidden;
  padding: 0;
  position: absolute;
  white-space: nowrap;
  width: 1px;
}

.c12 {
  text-align: left;
}

.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  font-size: 1rem;
  padding: 2rem;
  color: #FFFFFF;
  background-color: #2C0230;
}

.c9 {
  margin-top: 1rem;
}

.c9 button {
  border-color: #2C0230;
  width: 100%;
  font-size: 1rem;
}

.c11 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  margin-top: 1rem;
}

.c11 a {
  cursor: pointer;
  padding: 0 0.25rem;
  color: #FFFFFF;
  border-color: #FFFFFF;
}

.c11 a:hover {
  color: #FFFFFF;
  border-color: #FFFFFF;
}

.c3 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
}

.c2 {
  margin-bottom: 2rem;
  font-size: 2.625rem;
}

@media (min-width:740px) {
  .c7 {
    max-width: 290px;
  }
}

@media (min-width:740px) {
  .c9 button {
    width: auto;
  }
}

@media (min-width:1024px) {
  .c2 {
    font-size: 4rem;
  }
}

<div
  className="c0"
>
  <p
    className="c1 c2"
    color="inherit"
    size="s"
  >
    sign up letter
  </p>
  <form
    className="c3"
    onSubmit={[Function]}
  >
    <label
      className="c4"
    >
      <span
        className="c5 c6"
        color="inherit"
        size="s"
      >
        Please choose an option
      </span>
      <select
        className="c7"
        onChange={[Function]}
        placeholder="Please choose an option"
        value=""
      >
        <option
          disabled={true}
        >
          Please choose an option
        </option>
        <option
          value=""
        >
          -- Select age group --
        </option>
        <option
          value="Option one"
        >
          The first option
        </option>
        <option
          value="Option two"
        >
          The second option
        </option>
        <option
          value="Option three"
        >
          The third option
        </option>
        <option
          value="Option four"
        >
          The fourth option
        </option>
      </select>
      
    </label>
    <p
      className="c8"
      color="inherit"
      size="m"
    >
      Now please select your teaching group so you get the right updates.
    </p>
    <div
      className="c9"
    >
      <button
        className="c10"
        color="teal"
        href="/#"
        onClick={[Function]}
        target="_self"
        type="button"
      >
        Subscribe
      </button>
    </div>
  </form>
  <div
    className="c11"
  >
    <div
      className="c12"
      dangerouslySetInnerHTML={
        Object {
          "__html": "check <a href=\\"https://www.comicrelief.com/privacy-notice\\">Privacy policy</a>",
        }
      }
    />
  </div>
</div>
`;

exports[`renders correctly 1`] = `
.c1 {
  font-size: 1rem;
  text-transform: uppercase;
  font-weight: 400;
  font-family: 'Anton',Impact,sans-serif;
}

.c8 {
  font-size: 1rem;
  text-transform: inherit;
  font-weight: bold;
  font-family: 'Montserrat',Helvetica,Arial,sans-serif;
}

.c12 {
  display: inline-block;
  padding: 1rem 1.5rem;
  text-align: center;
  -webkit-text-decoration: none;
  text-decoration: none;
  font-weight: 700;
  border-radius: 2rem;
  cursor: pointer;
  background-color: #22CFD8;
  color: #2C0230;
}

.c12:hover {
  background-color: #86E4E9;
  color: #2C0230;
}

.c10 {
  font-family: 'Montserrat',Helvetica,Arial,sans-serif;
  font-weight: 400;
  position: relative;
  box-sizing: border-box;
  width: 100%;
  padding: 13px;
  margin: 10px 0;
  font-size: 1.25rem;
  background-color: #FFFFFF;
  border: 1px solid;
  border-color: #2C0230;
  box-shadow: none;
  -webkit-appearance: none;
  -moz-appearance: none;
  appearance: none;
  color: #2C0230;
}

.c10:focus {
  border: 1px solid #666;
}

.c10:focus::-webkit-input-placeholder {
  color: #666;
}

.c10:focus::-moz-placeholder {
  color: #666;
}

.c10:focus:-ms-input-placeholder {
  color: #666;
}

.c10:focus::placeholder {
  color: #666;
}

.c6 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
}

.c9 {
  visibility: border:0;
  -webkit-clip: rect(0 0 0 0);
  clip: rect(0 0 0 0);
  -webkit-clip-path: inset(50%);
  clip-path: inset(50%);
  height: 1px;
  margin: -1px;
  overflow: hidden;
  padding: 0;
  position: absolute;
  white-space: nowrap;
  width: 1px;
}

.c4 {
  text-align: left;
}

.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  font-size: 1rem;
  padding: 2rem;
  color: #FFFFFF;
  background-color: #2C0230;
}

.c3 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  margin-bottom: 0;
}

.c11 {
  margin-top: 1rem;
}

.c11 button {
  border-color: #2C0230;
  width: 100%;
  font-size: 1rem;
}

.c13 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  margin-top: 1rem;
}

.c13 a {
  cursor: pointer;
  padding: 0 0.25rem;
  color: #FFFFFF;
  border-color: #FFFFFF;
}

.c13 a:hover {
  color: #FFFFFF;
  border-color: #FFFFFF;
}

.c5 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
}

.c7 {
  width: 100%;
}

.c2 {
  margin-bottom: 2rem;
  font-size: 2.625rem;
}

@media (min-width:740px) {
  .c10 {
    max-width: 290px;
  }
}

@media (min-width:740px) {
  .c11 button {
    width: auto;
  }
}

@media (min-width:1024px) {
  .c2 {
    font-size: 4rem;
  }
}

<div
  className="c0"
>
  <p
    className="c1 c2"
    color="inherit"
    size="s"
  >
    sign up letter
  </p>
  <div
    className="c3"
  >
    <div
      className="c4"
      dangerouslySetInnerHTML={
        Object {
          "__html": "<h1> Top Copy</h1> <p>Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua. Ut enim ad minim veniam, quis nostrud exercitation ullamco laboris nisi ut aliquip ex ea commodo consequat.</p>",
        }
      }
    />
  </div>
  <form
    className="c5"
    onSubmit={[Function]}
  >
    <label
      className="c6 c7"
      htmlFor="email"
      name="email"
      onChange={[Function]}
      placeholder="example@youremail.com"
      value=""
    >
      <span
        className="c8 c9"
        color="inherit"
        size="s"
      >
        email
      </span>
      <input
        className="c10 c7"
        id="email"
        name="email"
        onChange={[Function]}
        placeholder="example@youremail.com"
        type="email"
        value=""
      />
      
    </label>
    <div
      className="c11"
    >
      <button
        className="c12"
        color="teal"
        href="/#"
        onClick={[Function]}
        target="_self"
        type="button"
      >
        Subscribe
      </button>
    </div>
  </form>
  <div
    className="c13"
  >
    <div
      className="c4"
      dangerouslySetInnerHTML={
        Object {
          "__html": "check <a href=\\"https://www.comicrelief.com/privacy-notice\\">Privacy policy</a>",
        }
      }
    />
  </div>
</div>
`;
